Firmware/Arduino/AdafruitRGBMatrix_EPXDriver.cpp

```diff
@@ -5,13 +5,83 @@
+/* ----------------------------------------------------------------------
+The RGB matrix must be wired to VERY SPECIFIC pins, different for each
+microcontroller board. This first section sets that up for a number of
+supported boards.
+------------------------------------------------------------------------- */
+#if defined(_VARIANT_MATRIXPORTAL_M4_) // MatrixPortal M4
+	uint8_t rgbPins[]  = {7, 8, 9, 10, 11, 12};
+	uint8_t addrPins[] = {17, 18, 19, 20};
+	uint8_t clockPin   = 14;
+	uint8_t latchPin   = 15;
+	uint8_t oePin      = 16;
+#elif defined(_VARIANT_FEATHER_M4_) // Feather M4 + RGB Matrix FeatherWing
+	uint8_t rgbPins[]  = {6, 5, 9, 11, 10, 12};
+	uint8_t addrPins[] = {A5, A4, A3, A2};
+	uint8_t clockPin   = 13;
+	uint8_t latchPin   = 0;
+	uint8_t oePin      = 1;
+#elif defined(__SAMD51__) // M4 Metro Variants (Express, AirLift)
+	uint8_t rgbPins[]  = {6, 5, 9, 11, 10, 12};
+	uint8_t addrPins[] = {A5, A4, A3, A2};
+	uint8_t clockPin   = 13;
+	uint8_t latchPin   = 0;
+	uint8_t oePin      = 1;
+#elif defined(_SAMD21_) // Feather M0 variants
+	uint8_t rgbPins[]  = {6, 7, 10, 11, 12, 13};
+	uint8_t addrPins[] = {0, 1, 2, 3};
+	uint8_t clockPin   = SDA;
+	uint8_t latchPin   = 4;
+	uint8_t oePin      = 5;
+#elif defined(NRF52_SERIES) // Special nRF52840 FeatherWing pinout
+	uint8_t rgbPins[]  = {6, A5, A1, A0, A4, 11};
+	uint8_t addrPins[] = {10, 5, 13, 9};
+	uint8_t clockPin   = 12;
+	uint8_t latchPin   = PIN_SERIAL1_RX;
+	uint8_t oePin      = PIN_SERIAL1_TX;
+#elif defined(ESP32)
+  // 'Safe' pins, not overlapping any peripherals:
+  // GPIO.out: 4, 12, 13, 14, 15, 21, 27, GPIO.out1: 32, 33
+  // Peripheral-overlapping pins, sorted from 'most expendible':
+  // 16, 17 (RX, TX)
+  // 25, 26 (A0, A1)
+  // 18, 5, 9 (MOSI, SCK, MISO)
+  // 22, 23 (SCL, SDA)
+  uint8_t rgbPins[]  = {4, 12, 13, 14, 15, 21};
+  uint8_t addrPins[] = {16, 17, 25, 26};
+  uint8_t clockPin   = 27; // Must be on same port as rgbPins
+  uint8_t latchPin   = 32;
+  uint8_t oePin      = 33;
+#elif defined(ARDUINO_TEENSY40)
+  uint8_t rgbPins[]  = {15, 16, 17, 20, 21, 22}; // A1-A3, A6-A8, skip SDA,SCL
+  uint8_t addrPins[] = {2, 3, 4, 5};
+  uint8_t clockPin   = 23; // A9
+  uint8_t latchPin   = 6;
+  uint8_t oePin      = 9;
+#elif defined(ARDUINO_TEENSY41)
+  uint8_t rgbPins[]  = {26, 27, 38, 20, 21, 22}; // A12-14, A6-A8
+  uint8_t addrPins[] = {2, 3, 4, 5};
+  uint8_t clockPin   = 23; // A9
+  uint8_t latchPin   = 6;
+  uint8_t oePin      = 9;
+#endif
+
+
+CAdafruitRGBMatrix_EPXDriver::CAdafruitRGBMatrix_EPXDriver(int width, int height)
 	: CLEDDriverBase(width * height)
 {
 	m_width = width;
 	m_height = height;
 	m_nRainbowIteration = 0;
-	m_pRGBmatrixPanel = new RGBmatrixPanel(pinA, pinB, pinC, pinD, pinCLK, pinOE, pinLAT, false, width, prgbpins);
+	brightness = 255;
+	m_pRGBmatrixPanel = new Adafruit_Protomatter(
+		64,							// Width of matrix (or matrix chain) in pixels
```

Comment:
Could we specify 'width' here instead of the hardcoded number 64?

Reply:
Yep! Updated to use the `width` passed into the constructor in commit 99b97db